crates/turborepo-ffi/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+mod proto {
+    include!(concat!(env!("OUT_DIR"), "/_.rs"));
+}
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct Buffer {
+    len: u32,
+    data: *mut u8,
+}
+
+impl<T: prost::Message> From<T> for Buffer {
+    fn from(value: T) -> Self {
+        let mut bytes = value.encode_to_vec();
+        let data = bytes.as_mut_ptr();
+        let len = bytes.len() as u32;
+        std::mem::forget(bytes);
```

Comment:
This feels off to me. If I'm understanding `mem::forget` this means `Buffer::from(message)` assumes the constructed `data` will be passed to another language that will free the data once it's used. Would it be safer to forget the right before we pass it back?

Maybe [`ManuallyDrop`](https://doc.rust-lang.org/std/mem/struct.ManuallyDrop.html) would be good here to signal this expectation?